demo/api/src/user-permission/user-permission-config.service.ts

```diff
@@ -0,0 +1,65 @@
+import {
+    ContentScopes,
+    ContentScopesForUser,
+    FindUsersArgs,
+    Permissions,
+    PermissionsForUser,
+    User,
+    UserPermissionConfigInterface,
+    UserPermissions,
+    Users,
+} from "@comet/cms-api";
+import { Injectable } from "@nestjs/common";
+
+const staticUsers: User[] = [
+    {
+        id: "1",
+        name: "Admin",
+        email: "demo@comet-dxp.com",
+        language: "en",
+    },
+    {
+        id: "2",
+        name: "Non-Admin",
+        email: "test@test.com",
+        language: "en",
+    },
+];
+
+@Injectable()
+export class UserPermissionsConfigService implements UserPermissionConfigInterface {
```

Comment:
naming:

For getAvailablePermissions / getAvailableContentScopes "config" is ok.

I wouldn't name a service that provides users and permissions as "config".

- UserPermissionsUserProviderService
- UserPermissionsProviderService
- UserPermissionsUserService

and move getAvailablePermissions/getPermissionsForUser directly to module config? (Or do you have a use case in mind where this needs to be a service as well?)

Reply:
The naming now is `UserPermissionsOptions` which is in line to other NestJS-Modules.

Reply:
Sorry, but this was a step backwards.

- having a module config/options is Ok (Options is the better name)
- I would like to keep the service (for easy DI)
- -> we need both: Options that contain a service. And this service I would not name Config (or Options)
- I propose the revert the last commit and rename `UserPermissionsConfigService` -> `UserPermissionsUserService` (and `UserPermissionConfigInterface` -> `UserPermissionUserInterface`)
- and move getAvailablePermissions/getAvailableContentScopes out of the service into the module options as availablePermissions/availableContentScopes option

Reply:
There is now a UserPermissionsUserService in the options. The options itself can be passed via useExisting/useFactory/useClass (the demo now uses useFactory and passes the UserService which is provided by the AuthModule). However, the UserService itself has to be passed as instance. Nevertheless I think that this version offers enough flexibility to handle all use cases.